src/Shared/OSMModels/OsmBaseObject.swift

```diff
@@ -531,9 +531,21 @@ class OsmBaseObject: NSObject, NSCoding, NSCopying {
 		"living_street": .name
 	]
 	func givenName() -> String? {
-		if let name = tags["name"] {
+		//first try name tag
+        if let name = tags["name"] {
 			return name
 		}
+        //then try name:en
+        if let name = tags["name:en"] {
+            return name
+        }
+        //then try any other name:* tag
+        if let name = tags.filter({ key, _ in
+            key.starts(with: "name:")
+        }).first?.value {
+            return name
```

Comment:
Could you replace the filter/first pair with first(where:)? Thanks!

Reply:
done!